platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -2467,18 +2486,29 @@ namespace platoon_strategic_ihp
             request.location = pose_to_ecef(pose_msg_);
             mobility_request_publisher_(request);
             ROS_DEBUG_STREAM("Published Mobility Candidate-Join request to the leader");
-            pm_.current_plan = PlatoonPlan(true, currentTime, planId, pm_.targetLeaderId);
-            ROS_DEBUG_STREAM("current plan peer id: " << pm_.targetLeaderId);
+            ROS_DEBUG_STREAM("current plan peer id: " << pm_.current_plan.peerId);
+
+            // Update the local record of the new activity plan and now establish that we have a platoon plan as well,
+            // which allows us to start sending necessary op STATUS messages
+            pm_.current_plan = ActionPlan(true, currentTime, planId, pm_.current_plan.peerId);
+            pm_.currentPlatoonID = planId;
+            if (pm_.targetPlatoonID.length() > 0)
+            {
+                pm_.currentPlatoonID = pm_.targetPlatoonID;
+                pm_.targetPlatoonID = "";
+            }
+            pm_.platoonLeaderID = pm_.current_plan.peerId;
         }
 
-        ROS_DEBUG_STREAM("pm_.getTotalPlatooningSize()" << pm_.getTotalPlatooningSize());
         //Task 4: publish platoon status message (as single joiner)
-        if (pm_.getTotalPlatooningSize() > 1) {
+        if (pm_.current_plan.valid) //don't want to do this until iterations after MobReq message is delivered above
```

Comment:
I dont understand why we have this condition. I thought in this new version the current plan is only for joining. But according to this condition, we wont get any status msgs during regular operation

Reply:
There is a lot to keep track of with the layers of changing states! I may very well have overlooked something, but I think this is good. For the first several calls to run_candidate_follower(), current_plan.valid will be false and this block will not be executed. Presumably host is moving into position during that time. At some point gap is adequate, and the "task 3" block runs, which sends the MobReq and sets current_plan.valid. From then on, while still in CANDIDATEFOLLOWER state, host will send op STATUS messages here. Once an ACK is received, host transitions to FOLLOWER and starts sending op STATUS from run_follower().